### PR TITLE
Update doc for CA1801 to mention exclusion of serialization methods

### LINKED ISSUE #4987: CA1801 - Serialization methods are now excluded
Hi there,

With the release of fxcop 3.X the analyzer will no longer report on serialization methods unused parameters as there is no way to fix them.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5030cf5c-ae08-5a08-0b47-995006379292
* Version Independent ID: 9e1df70a-127c-eafc-e027-360797faf3a6
* Content: [CA1801: Review unused parameters - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1801?view=vs-2019)
* Content Source: [docs/code-quality/ca1801.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/code-quality/ca1801.md)
* Product: **visual-studio-windows**
* Technology: **vs-ide-code-analysis**
* GitHub Login: @Mikejo5000
* Microsoft Alias: **mikejo**

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #4987

### CODE REVIEW

**Comment by PRMerger16** [comment]
@Evangelink : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/code-quality/ca1801.md [hunk 1]
```diff
@@ -35,6 +35,10 @@ This rule does not examine the following kinds of methods:
 
 - Methods used as event handlers.
 
+- Serialization constructors.
+
+- Serialization `GetObjectData` method.
```
Comment (mavasani) [inline docs/code-quality/ca1801.md:40]: Should we add a hyperlink to https://docs.microsoft.com/dotnet/api/system.runtime.serialization.iserializable.getobjectdata?

File: docs/code-quality/ca1801.md [hunk 2]
```diff
@@ -35,6 +35,10 @@ This rule does not examine the following kinds of methods:
 
 - Methods used as event handlers.
 
+- Serialization constructors.
```
Comment (mavasani) [inline docs/code-quality/ca1801.md:38]: Can we add hyperlink to base type/interface or some other generic documentation page about serialization constructors?

**Comment by ktoliver** [comment]
Hi @Evangelink - Could you respond to the questions when you have a chance? Thanks.

**Comment by PRMerger13** [comment]
@Evangelink : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/code-quality/ca1801.md [hunk 3]
```diff
@@ -35,6 +35,10 @@ This rule does not examine the following kinds of methods:
 
 - Methods used as event handlers.
 
+- Serialization constructors ([see guidelines](https://docs.microsoft.com/fr-fr/dotnet/standard/serialization/serialization-guidelines#supporting-runtime-serialization)).
+
+- Serialization <xref:System.Runtime.Serialization.ISerializable.GetObjectData> method.
```
Comment (Evangelink) [inline docs/code-quality/ca1801.md:40]: Please review the syntax.

File: docs/code-quality/ca1801.md [hunk 4]
```diff
@@ -35,6 +35,10 @@ This rule does not examine the following kinds of methods:
 
 - Methods used as event handlers.
 
+- Serialization constructors ([see guidelines](https://docs.microsoft.com/fr-fr/dotnet/standard/serialization/serialization-guidelines#supporting-runtime-serialization)).
```
Comment (Evangelink) [inline docs/code-quality/ca1801.md:38]: Is it ok to point to a documentation page like this?

File: docs/code-quality/ca1801.md [hunk 5]
```diff
@@ -35,6 +35,10 @@ This rule does not examine the following kinds of methods:
 
 - Methods used as event handlers.
 
+- Serialization constructors ([see guidelines](https://docs.microsoft.com/dotnet/standard/serialization/serialization-guidelines#supporting-runtime-serialization)).
```
Comment (Evangelink) [inline docs/code-quality/ca1801.md:38]: Is it ok to use this syntax?

**Comment by PRMerger14** [comment]
@Evangelink : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/code-quality/ca1801.md (hunk 5 above)
Comment (Mikejo5000) [reply docs/code-quality/ca1801.md:38]: It's OK to use. Best syntax for a docs link like this is: /dotnet/standard/serialization/serialization-guidelines#supporting-runtime-serialization

File: docs/code-quality/ca1801.md (hunk 3 above)
Comment (Mikejo5000) [reply docs/code-quality/ca1801.md:40]: Looks like you already fixed these and removed "fr-fr".

Comment (Mikejo5000) [reply docs/code-quality/ca1801.md:40]: xref looks fine

File: docs/code-quality/ca1801.md (hunk 5 above)
Comment (Mikejo5000) [inline docs/code-quality/ca1801.md:38]:
```suggestion
- Serialization constructors ([see guidelines](/dotnet/standard/serialization/serialization-guidelines#supporting-runtime-serialization)).
```

**Comment by PRMerger12** [comment]
@Mikejo5000 : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by Mikejo5000** [comment]
#sign-off

**Pull request merged**